src/main/java/com/yugabyte/sample/apps/SqlUpdatesWithProcedures.java

```diff
@@ -0,0 +1,275 @@
+
+  public SqlUpdatesWithProcedures() {
+    buffer = new byte[appConfig.valueSize];
+  }
+
+  @Override
+  public void createTablesIfNeeded() throws Exception {
+    // Check that (extra_ required options are set (maxWrittenKey and loadTesterUUID) set.
+    if (appConfig.maxWrittenKey <= 0) {
+      LOG.fatal("Workload requires option --max_written_key to be set. \n " +
+                    "Run 'java -jar yb-sample-apps.jar --help SqlUpdatesWithProcedures' for more details");
+      System.exit(1);
+    }
+    if (CmdLineOpts.loadTesterUUID == null) {
+      LOG.fatal("Workload requires option --uuid to be set. \n " +
+                    "Run 'java -jar yb-sample-apps.jar --help SqlUpdatesWithProcedures' for more details");
+      System.exit(1);
+    }
+
+    // Expecting that the SqlInserts ap was already run and created the table.
+    // This app just reuses the table and updates the values for the existing rows.
+    Connection connection = getPostgresConnection();
+    DatabaseMetaData dbm = connection.getMetaData();
+    ResultSet tables = dbm.getTables(null, null, getTableName(), null);
+    if (!tables.next()) {
+      LOG.fatal(
+          String.format("Could not find the %s table. Did you run SqlInserts first?" +
+                            "Run 'java -jar yb-sample-apps.jar --help SqlUpdatesWithProcedures' for more details",
+                        getTableName()));
+      System.exit(1);
+    }
+  }
+
+  public String getTableName() {
+    String tableName = appConfig.tableName != null ? appConfig.tableName : DEFAULT_TABLE_NAME;
+    return tableName.toLowerCase();
+  }
+
+  @Override
+  public void createStoredProcedures() throws Exception {
+    // Drop previous procedures.
+    getPostgresConnection().createStatement().execute(
+      String.format("DROP PROCEDURE IF EXISTS %s;", getStoredProcedureName()));
+
+    StringBuilder sb = new StringBuilder();
+    sb.append("CREATE PROCEDURE ").append(getStoredProcedureName()).append("(");
+    for (int i = 0; i < getBatchSize(); i++) {
+      if (i > 0) {
+        sb.append(", ");
+      }
+      sb.append("k").append(i).append(" text, v").append(i).append(" text");
+    }
+    sb.append(") AS '");
+    for (int i = 0; i < getBatchSize(); i++) {
+      sb.append("UPDATE ").append(getTableName())
+        .append(" SET v=v").append(i)
+        .append(" WHERE k=k").append(i).append(";");
+    }
+    sb.append("' LANGUAGE sql;");
+    getPostgresConnection().createStatement().execute(sb.toString());
+  }
+
+  public String getStoredProcedureName() {
+    return appConfig.storedProcedureName;
+  }
+
+  public int getBatchSize() {
+    return appConfig.updateBatchSize;
+  }
+
+  private PreparedStatement getPreparedSelect() throws Exception {
+    if (preparedSelect == null) {
+      preparedSelect = getPostgresConnection().prepareStatement(
+          String.format("SELECT k, v FROM %s WHERE k = ?;", getTableName()));
+    }
+    return preparedSelect;
+  }
+
+  @Override
```

Comment:
Do we need the reads as part of the test?

Reply:
I have the reads since we have those in `SqlUpdates` as well, but they are only really there to verify that we have actually updated the rows